include/alpaka/mem/buf/Traits.hpp

```diff
@@ -133,9 +133,12 @@ namespace alpaka
     //! \param extent The extent of the buffer.
     //! \return The newly allocated buffer.
     template<typename TPlatform, typename TElem, typename TIdx, typename TExtent>
-    ALPAKA_FN_HOST auto allocMappedBuf(DevCpu const& host, TExtent const& extent = TExtent())
+    ALPAKA_FN_HOST auto allocMappedBuf(
+        DevCpu const& host,
+        TPlatform const& platform = TPlatform(),
```

Comment:
I do not know if allowing a default value here is a good idea or not.
In the SYCL case, what does the default constructor of the platform do?

Reply:
After some more thoughts, I think it would be better to _not_ have the default:
  - the SYCL backend requires a valid platform object
  - code written to be portable should be compatible with all backends

Reply:
I agree, platforms cannot be relied upon to be default constructible.